pkg/route/api/validation/validation.go

```diff
@@ -76,6 +79,60 @@ func ValidateRouteStatusUpdate(route *routeapi.Route, older *routeapi.Route) fie
 	return allErrs
 }
 
+// ExtendedValidateRoute performs an extended validation on the route
+// including checking that the TLS config is valid.
+func ExtendedValidateRoute(route *routeapi.Route) field.ErrorList {
+	tlsConfig := route.Spec.TLS
+	result := field.ErrorList{}
+
+	tlsFieldPath := field.NewPath("spec").Child("tls")
+	if errs := validateTLS(route, tlsFieldPath); len(errs) != 0 {
+		result = append(result, errs...)
+	}
+
+	// TODO: Check if we can be stricter with validating the certificate
+	//       is for the route hostname. Don't want existing routes to
+	//       break, so disable the hostname validation for now.
+	// hostname := route.Spec.Host
+	hostname := ""
+	var certPool *x509.CertPool
+
+	if len(tlsConfig.CACertificate) > 0 {
+		certPool = x509.NewCertPool()
+		if ok := certPool.AppendCertsFromPEM([]byte(tlsConfig.CACertificate)); !ok {
+			result = append(result, field.Invalid(tlsFieldPath.Child("caCertificate"), tlsConfig.CACertificate, "failed to parse CA certificate"))
+		}
+	}
+
+	verifyOptions := &x509.VerifyOptions{
+		DNSName: hostname,
+		Roots:   certPool,
+	}
+
+	if len(tlsConfig.Certificate) > 0 {
+		if _, err := validateCertificatePEM(tlsConfig.Certificate, verifyOptions); err != nil {
+			result = append(result, field.Invalid(tlsFieldPath.Child("certificate"), tlsConfig.Certificate, err.Error()))
+		}
+	}
+
+	if len(tlsConfig.Key) > 0 {
```

Comment:
Yeah, we just concatenate the certificate and key pem data to a file and then use that in the haproxy router. Since haproxy is reading this (and other certs) from a directory, it would present the first certificate for that host + I think its uses the first key. 
That's a good idea to just concatenate the cert and key data and pass it as both the key and cert pem files. Thx